src/InitialConditions/initial_conditions.jl

```diff
@@ -0,0 +1,763 @@
+
+The `InitialCondition` described in [Ackerman2009](@cite), but with a
+hydrostatically balanced pressure profile.
+"""
+struct DYCOMS_RF02 <: InitialCondition end
+
+"""
+    GABLS
+
+The `InitialCondition` described in [???](@cite), but with a hydrostatically
+balanced pressure profile.
+"""
+struct GABLS <: InitialCondition end
+
+for IC_Type in (:Dycoms_RF01, :Dycoms_RF02, :GABLS)
+    IC_Type_caps = Symbol(uppercase(string(IC_Type)))
+    θ_func_name = Symbol(IC_Type, :_θ_liq_ice)
+    q_tot_func_name = Symbol(IC_Type, :_q_tot)
+    u_func_name = Symbol(IC_Type, IC_Type == :Dycoms_RF01 ? :_u0 : :_u)
+    v_func_name = Symbol(IC_Type, IC_Type == :Dycoms_RF01 ? :_v0 : :_v)
+    tke_func_name = Symbol(IC_Type, :_tke)
+    @eval function (initial_condition::$IC_Type_caps)(params)
+        FT = eltype(params)
+        thermo_params = CAP.thermodynamics_params(params)
+        p_0 = initial_surface_pressure(initial_condition, thermo_params)
+        θ = APL.$θ_func_name(FT)
+        q_tot = APL.$q_tot_func_name(FT)
+        p = hydrostatic_pressure_profile(; thermo_params, p_0, θ, q_tot)
+        u = APL.$u_func_name(FT)
+        v = APL.$v_func_name(FT)
+        tke = APL.$tke_func_name(FT)
+        function local_state(local_geometry)
+            (; z) = local_geometry.coordinates
+            return LocalState(;
+                params,
+                geometry = local_geometry,
+                thermo_state = TD.PhaseEquil_pθq(
+                    thermo_params,
+                    p(z),
+                    θ(z),
+                    q_tot(z),
+                ),
+                velocity = Geometry.UVVector(u(z), v(z)),
+                turbconv_state = EDMFState(; tke = tke(z)),
+            )
+        end
+        return local_state
+    end
+end
+
+"""
+    Rico
+
+The `InitialCondition` described in [???](@cite), but with a hydrostatically
+balanced pressure profile.
+"""
+struct Rico <: InitialCondition end
+
+function (initial_condition::Rico)(params)
+    FT = eltype(params)
+    thermo_params = CAP.thermodynamics_params(params)
+    p_0 = initial_surface_pressure(initial_condition, thermo_params)
+    θ = APL.Rico_θ_liq_ice(FT)
+    q_tot = APL.Rico_q_tot(FT)
+    p = hydrostatic_pressure_profile(; thermo_params, p_0, θ, q_tot)
+    u = APL.Rico_u(FT)
+    v = APL.Rico_v(FT)
+
+    # Set tke to a function of the inversion height (i.e., the boundary layer
+    # height), denoted here by h. Since the free convective limit does not
+    # apply, the inversion layer height must be computed from Ri_bulk.
+    # TODO: Check this and cite the relevant paper.
+    grav = TD.Parameters.grav(thermo_params)
+    θ_virt(z) = TD.virtual_pottemp(
+        thermo_params,
+        TD.PhaseEquil_pθq(thermo_params, p(z), θ(z), q_tot(z)),
+    )
+    θ_virt_0 = θ_virt(FT(0))
+    Ri_bulk(z) =
+        grav * (θ_virt(z) - θ_virt_0) * z / θ_virt_0 / (u(z)^2 + v(z)^2)
+    Ri_bulk_crit = FT(0.2)
+    z_values = FT(0):FT(100):FT(30e3)
```

Comment:
Let's not hard-code numbers for zmax here

Reply:
We know that the target z will be in the first few kilometers above the surface, so it's pretty safe to add an upper bound of tens of kilometers. Also, we should probably not make the initial conditions depend on the model configuration.

Reply:
@trontrytel's voted to keep this as is. Let's roll with it.